packages/@magic-sdk/react-native-bare/src/react-native-sdk-base.ts

```diff
@@ -1,7 +1,70 @@
-import { SDKBase } from '@magic-sdk/provider';
+import { createPromiEvent, MagicSDKAdditionalConfiguration, SDKBase, UserModule } from '@magic-sdk/provider';
+import AsyncStorage from '@react-native-async-storage/async-storage';
 import { ReactNativeWebViewController } from './react-native-webview-controller';
 
+type UserLoggedOutCallback = (loggedOut: boolean) => void;
+
 export class SDKBaseReactNative extends SDKBase {
+  // this UserModule instance used for the actual calls to
+  // `isLoggedIn` and `logOut`, given that we assign new functions
+  // to `this.user.isLoggedIn` and `this.user.logOut`.
+  private usr: UserModule;
+  private userLoggedOutCallbacks: UserLoggedOutCallback[] = [];
+
+  constructor(public readonly apiKey: string, options?: MagicSDKAdditionalConfiguration) {
+    super(apiKey, options);
+    this.usr = new UserModule(this);
+    if (options?.useStorageCacheMobile) {
```

Comment:
Can this new API be added to the BaseSDK? So that web SDK may leverage this as well.

https://github.com/magiclabs/magic-js/blob/d88ee684e865b410e0c398ae84eebcb026fbb6ef/packages/%40magic-sdk/provider/src/core/sdk.ts#L174